R/explore.R

```diff
@@ -121,6 +121,220 @@ mc_plot_univariate <- function(i, x, dist="ZINB") {
+        ggplot2::theme_minimal() +
+        ggplot2::ylab("Total Moose") +
+        ggplot2::xlab(i)
+
+      tot <- stats::aggregate(dat[, c("y", "zhat")], list(z = dat$z), mean)
+      tot$z_int <- as.integer(tot$z)
+
+      if(interactive) {
+        out <- dplyr::group_by(dat, .data$z)
+        out <- dplyr::mutate(out,
+                             IQR = stats::IQR(.data$y),
+                             q75 = stats::quantile(.data$y, 0.75),
+                             q25 = stats::quantile(.data$y, 0.25))
+        out <- dplyr::filter(out,
+                             .data$y > (.data$q75 + (1.5 * .data$IQR)) |
+                               .data$y < (.data$q25 - (1.5 * .data$IQR)))
+        out <- dplyr::mutate(out, tt = paste0("SU_ID: ", .data$SU_ID,
+                                              "<br>", .env$i, ": ", .data$z,
+                                              "<br>Total Moose: ", .data$y))
+        dat <- dplyr::mutate(dat, tt = .data$z)
+        tot <- dplyr::mutate(tot, tt = paste0(.data$z,
+                                              "<br>Mean: ", round(.data$zhat, 2)))
+
+        p <- p +
+          ggiraph::geom_boxplot_interactive(
+            data = dat, fill = "grey", outlier.shape = NA, # omit outliers
+            ggplot2::aes(tooltip = .data$tt, data_id = .data$tt)) +
+          ggiraph::geom_point_interactive(                 # manually add outliers
+            data = out, size = 3, alpha = 0.75,
+            ggplot2::aes(tooltip = .data$tt, data_id = .data$tt)) +
+          ggiraph::geom_point_interactive(
+            data = tot, size = 3, col = 4,
+            ggplot2::aes(tooltip = .data$tt, data_id = .data$tt))
+      } else {
+        p <- p +
+          ggplot2::geom_boxplot(fill = "grey")
+      }
+
+      # Lines at the end for all plots, never interactive
+      p <- p +
+        ggplot2::geom_line(
+          ggplot2::aes(x = .data$z_int, y = .data$zhat),
+          data = tot, col = 4)
+
+    } else {
+      p <- ggplot2::ggplot(
+        dat,
+        ggplot2::aes(
+          y = .data$y,
+          x = .data$z)) +
+        ggplot2::geom_line(ggplot2::aes(x = .data$z, y = .data$zhat), col = 4) +
+        ggplot2::theme_minimal() +
+        ggplot2::ylab("Total Moose") +
+        ggplot2::xlab(i)
+
+      dat <- dplyr::mutate(dat, tt = paste0("SU_ID: ", .data$SU_ID,
+                                            "<br>", .env$i, ": ", round(.data$z, 2),
+                                            "<br>Total Moose: ", .data$y))
+
+      if(interactive){
+        p <- p + ggiraph::geom_point_interactive(
+          data = dat, size = 3, alpha = 0.75,
+          ggplot2::aes(tooltip = .data$tt, data_id = .data$tt))
+      } else {
+        p <- p + ggplot2::geom_point()
+      }
+
+    }
+  }
+  if (type == "map") {
+    p <- ggplot2::ggplot(data = x,
+                         ggplot2::aes(x = .data$CENTRLON, y = .data$CENTRLAT,
+                                      fill = .data[[i]])) +
```

Comment:
Use the `.data[[i]]` notation instead of `aes_string` (I think that's less recommended in recent years, but not sure if it's going anywhere).